tests/neg/i12344.scala

```diff
@@ -0,0 +1,23 @@
+import scala.quoted.*
+
+class C(using q: Quotes)(i: Int = 1, f: q.reflect.Flags = q.reflect.Flags.EmptyFlags)
+
+def test1a(using q: Quotes) = new C()                           // error
+def test2a(using q: Quotes) = new C(1)                          // error
+def test3a(using q: Quotes) = new C(1, q.reflect.Flags.Lazy)    // error
+def test4a(using q: Quotes) = new C(f = q.reflect.Flags.Lazy)   // error
+
+def test1b(using q: Quotes) = C()                               // error
```

Comment:
It seems like it should compile.

Reply:
As you mentioned in https://github.com/lampepfl/dotty/issues/12344#issuecomment-833351641, for the default parameters to apply, we do need double `()`:

```Scala
def test1e(using q: Quotes) = new C()()
def test2e(using q: Quotes) = C()()
```

Reply:
That goes against the design philosophy of `using` clauses though doesn't it? My understanding is that if `using` isn't specified at call-site then it shouldn't map to the `using` param clause. Having the first `()` map to the `using q` clause breaks that rule.

Reply:
@japgolly I think there is no such mapping. It seems the ctor gets an initial empty param list that is not written, see the linked comment. The first explicit params in the call correspond to that one. This is perhaps more natural in Scala 2 where you expect auto-application and only one implicit at the end, so `new C` makes sense as `new C()(implicits)`. I see it must still be explicit in Scala 3 for case class: "A case class must have at least one non-implicit parameter list".

Reply:
Given a definition like the following:

```Scala
class C(using Int)
object C
```

Without prepending the constructor with`()`, the identifier `C` would be ambiguous (Scala 3 generailzes synthetic `apply` to [all classes][1] to avoid `new`).

For the example reported in #12344, the implicit parameter list comes after the `synthetic` empty parameter list `()`, and finally follows the default parameter list.

[1]: http://dotty.epfl.ch/docs/reference/other-new-features/creator-applications.html